src/main/java/com/avispl/symphony/dal/communicator/aggregator/GoogleWorkspaceCommunicator.java

```diff
@@ -920,25 +953,23 @@ private void mapMonitoringProperty(Map<String, String> mappingStatistic, Map<Str
 	 * @param dynamic The dynamic map to populate with the mapped properties.
 	 */
 	private void mapDynamicStatistic(Map<String, String> mappingStatistic, Map<String, String> stats, Map<String, String> dynamic) {
-		String value;
+		String value = getDefaultValueForNullData(mappingStatistic.get(GoogleWorkspaceConstant.CPU_TEMPERATURE));
 		String name;
 		String propertyName;
-		JsonNode jsonNodeValue;
-		for (DynamicStatisticsDefinition dynamicItem : DynamicStatisticsDefinition.values()) {
-			name = dynamicItem.getName();
-			value = getDefaultValueForNullData(mappingStatistic.get(name));
-			propertyName = GoogleWorkspaceConstant.CPU_TEMPERATURE_GROUP + name;
-
-			jsonNodeValue = getJsonNodeValue(value);
-			if (jsonNodeValue != null) {
-				JsonNode lastObject = jsonNodeValue.get(jsonNodeValue.size() - 1).get(GoogleWorkspaceConstant.CPU_TEMPERATURE_INFO);
-				List<CPUTemperature> cpuTemperatures = convertJsonNodeToList(lastObject, new TypeReference<List<CPUTemperature>>() {
-				});
-				if (!cpuTemperatures.isEmpty()) {
-					String label = CPUTemperatureEnum.getByName(name).getValue();
-					int temperature = cpuTemperatures.stream().filter(cpuTemperature -> cpuTemperature.getLabel().equals(label)).map(CPUTemperature::getTemperature).findFirst()
-							.orElse(0);
+		JsonNode jsonNodeValue = getJsonNodeValue(value);
 
+		if (jsonNodeValue != null) {
+			JsonNode lastObject = jsonNodeValue.get(jsonNodeValue.size() - 1).get(GoogleWorkspaceConstant.CPU_TEMPERATURE_INFO);
```

Comment:
is it safe to do jsonNodeValue.size() - 1 here? 
the blocks only checks for the null value, is it possible that there's an empty array, so the result ends up being negative?

Reply:
> is it safe to do jsonNodeValue.size() - 1 here? the blocks only checks for the null value, is it possible that there's an empty array, so the result ends up being negative?

I added a condition to it